hyperhtml.js

```diff
@@ -193,321 +156,692 @@ var hyperHTML = (function () {'use strict';
+        update = setAttribute(target, removeAttributes);
         break;
-      case TEXT_NODE:
-        parent.textContent = child.textContent;
+      case 'text':
+        update = setTextContent(target);
+        break;
+      case 'virtual':
+        update = setVirtualContent(target, childNodes);
         break;
     }
+    return update;
   }
 
-  // it does exactly what it says
-  function removeNodeList(list, startIndex) {
-    var length = list.length, child;
-    while (startIndex < length--) {
-      child = list[length];
-      child.parentNode.removeChild(child);
+  // used for common path creation.
+  function Path(type, node) {
+    return {type: type, path: createPath(node)};
+  }
+
+  // ---------------------------------------------
+  // Hybrid Shims
+  // ---------------------------------------------
+
+  // WeakMap with partial EXPANDO fallback
+  var $WeakMap = typeof WeakMap === typeof $WeakMap ?
+      function () {
+        return {
+          get: function (obj) { return obj[EXPANDO]; },
+          set: function (obj, value) {
+            Object.defineProperty(obj, EXPANDO, {
+              configurable: true,
+              value: value
+            });
+          }
+        };
+      } :
+      WeakMap;
+
+  // Map with partial double Array fallback
+  var $Map = typeof Map === typeof $Map ?
+      function () {
+        var k = [], v = [];
+        return {
+          get: function (obj) {
+            return v[k.indexOf(obj)];
+          },
+          // being used with unique template literals
+          // there is never a case when a value is overwritten
+          // no need to check upfront for the indexOf
+          set: function (obj, value) {
+            v[k.push(obj) - 1] = value;
+          }
+        };
+      } :
+      Map;
+
+  // TODO: which browser needs these partial polyfills here?
+  var isArray = Array.isArray || function (a) { return a instanceof Array; };
+  var trim = EXPANDO.trim || function () { return this.replace(/^\s+|\s+$/g, ''); };
+
+  // ---------------------------------------------
+  // Shared variables
+  // ---------------------------------------------
+
+  // normalize Firefox issue with template literals
+  var templateObjects, unique;
+  if (FF) {
+    templateObjects = Object.create(null);
+    unique = function (template) {
+      var key = template.join(UID);
+      return templateObjects[key] ||
+            (templateObjects[key] = template);
+    };
+  }
+
+  // use native .append(...childNodes) where available
+  var appendNodes = 'append' in document ?
+      function (node, childNodes) {
+        node.append.apply(node, childNodes);
+      } :
+      function appendNodes(node, childNodes) {
+        for (var
+          i = 0,
+          length = childNodes.length;
+          i < length; i++
+        ) {
+          node.appendChild(childNodes[i]);
+        }
+      };
+
+  // returns children or retrieve them in IE/Edge
+  var getChildren = 'children' in document ?
```

Comment:
no, specially because that would return `true` being that inherited.

The `in` check is fine.